router/pkg/connectrpc/vanguard_service.go

```diff
@@ -0,0 +1,374 @@
+package connectrpc
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"connectrpc.com/connect"
+	"connectrpc.com/vanguard"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/dynamicpb"
+)
+
+// VanguardServiceConfig holds configuration for creating a Vanguard service
+type VanguardServiceConfig struct {
+	Handler     *RPCHandler
+	ProtoLoader *ProtoLoader
+	Logger      *zap.Logger
+}
+
+// VanguardService wraps the RPC handler and creates Vanguard services
+type VanguardService struct {
```

Comment:
We should have a better description on what this service is and what it does. The description of the wrapped package "connectrpc.com/vanguard" is pretty good.